TESTS/mbed_platform/error_handling/main.cpp

```diff
@@ -0,0 +1,365 @@
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_TYPE_CUSTOM, MBED_GET_ERROR_TYPE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_MODULE_PLATFORM, MBED_GET_ERROR_MODULE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_CODE_CREATE_FAILED, MBED_GET_ERROR_CODE(lastError));
+    mbed_error_status_t status = mbed_get_last_error_info( &error_ctx );
+    TEST_ASSERT(status == MBED_SUCCESS);
+    TEST_ASSERT_EQUAL_UINT(error_value, error_ctx.error_value);
+    
+    error_value = 0xAABBCC;
+    MBED_WARNING(MBED_ERROR_EACCES, "Posix Error", error_value );
+    lastError = mbed_get_last_error();
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_TYPE_POSIX, MBED_GET_ERROR_TYPE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_MODULE_UNKNOWN, MBED_GET_ERROR_MODULE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_CODE_EACCES, MBED_GET_ERROR_CODE(lastError));
+    status = mbed_get_last_error_info( &error_ctx );
+    TEST_ASSERT(status == MBED_SUCCESS);
+    TEST_ASSERT_EQUAL_UINT(error_value, error_ctx.error_value);
+    
+    error_value = 0;
+    error = MBED_MAKE_ERROR(MBED_MODULE_HAL, MBED_ERROR_CODE_UNKNOWN);
+    MBED_WARNING(error, "HAL Entity error", error_value );
+    lastError = mbed_get_last_error();
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_TYPE_SYSTEM, MBED_GET_ERROR_TYPE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_MODULE_HAL, MBED_GET_ERROR_MODULE(lastError));
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_CODE_UNKNOWN, MBED_GET_ERROR_CODE(lastError));
+    status = mbed_get_last_error_info( &error_ctx );
+    TEST_ASSERT(status == MBED_SUCCESS);
+    TEST_ASSERT_EQUAL_UINT(error_value, error_ctx.error_value);
+    
+    MBED_WARNING(MBED_ERROR_MUTEX_LOCK_FAILED, "Mutex lock failed", 0x4455 );
+    error = mbed_get_last_error();
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_MUTEX_LOCK_FAILED, error);
+    
+    error = mbed_get_first_error();
+    TEST_ASSERT_EQUAL_UINT(MBED_ERROR_OUT_OF_RESOURCES, error);
+    
+}
+
+/** Test error context capture
+ */
+void test_error_context_capture()
+{
+    uint32_t error_value = 0xABCD;
+    mbed_error_ctx error_ctx = {0};
+    
+    MBED_WARNING(MBED_ERROR_INVALID_ARGUMENT, "System type error", error_value );
+    mbed_error_status_t status = mbed_get_last_error_info( &error_ctx );
+    TEST_ASSERT(status == MBED_SUCCESS);
+    TEST_ASSERT_EQUAL_UINT(error_value, error_ctx.error_value);
+    TEST_ASSERT_EQUAL_UINT(osThreadGetId(), error_ctx.thread_id);
+    
+    //Capture thread info and compare
+    osRtxThread_t *current_thread = osRtxInfo.thread.run.curr;
+    TEST_ASSERT_EQUAL_UINT((uint32_t)current_thread->thread_addr, error_ctx.thread_entry_address);
+    TEST_ASSERT_EQUAL_UINT((uint32_t)current_thread->stack_size, error_ctx.thread_stack_size);
+    TEST_ASSERT_EQUAL_UINT((uint32_t)current_thread->stack_mem, error_ctx.thread_stack_mem);
+#ifdef MBED_CONF_ERROR_FILENAME_CAPTURE_ENABLED
+    TEST_ASSERT_EQUAL_STRING(MBED_FILENAME, error_ctx.error_filename);
+#endif    
+}
+
+#ifndef MBED_CONF_ERROR_HIST_DISABLED
```

Comment:
We should be using full words when possible. Is there a reason this needs to be shortened?